src/Collections/Sites.php

```diff
@@ -261,33 +287,40 @@ protected function getUuidByName(string $name): string
     }
 
     /**
-     * Looks up a site's UUID by its name.
+     * Returns the site UUID.
      *
-     * @param string $id Name of the site to look up
+     * @param string $site_id
+     *   The site name or UUID.
      * @return string
+     *   The site UUID.
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
+     * @throws \Pantheon\Terminus\Exceptions\TerminusNotFoundException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    protected function findUUIDByNameOrUUID($id): ?string
+    protected function getUuidBySiteId(string $site_id): ?string
     {
-        // If it LOOKS like a uuid, then we assume it is. Since a user is unlikely to name a site with this exact
-        // pattern this is a reasonably good test.
-        if ($this->isUUID($id)) {
-            return $id;
+        if (!$this->isValidUuidV4($site_id)) {
+            return $this->getUuidByName($site_id);
         }
-        return $this->getUuidByName($id);
+
+        $response = $this->request()->request(sprintf('sites/%s', $site_id), ['method' => 'get']);
+        if ($response->isError()) {
+            throw new TerminusNotFoundException($response->getData());
+        }
+
+        return $site_id;
     }
 
     /**
-     * Determine if the given string looks like a valid uuid.
+     * Returns TRUE if the string is a valid v4 UUID value.
      *
-     * This is not an exact test for uuids but it matches the general pattern:
-     *  xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
-     * where x is any hexidecimal character. This is close enough for our purposes.
+     * @param string $uuid
      *
-     * @param $id
      * @return bool
      */
-    protected function isUUID($id): bool
+    protected function isValidUuidV4(string $uuid): bool
     {
-        return preg_match('/[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}/', strtolower($id));
+        return preg_match('/^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/', strtolower($uuid));
```

Comment:
Do you know whether our platform always generates UUIDs that are strictly v4 conforming? It might be good to find the code that generates them to be sure. This is probably in ygg. Alternately, you probably wouldn't need to test too many actual UUIDs, if they are not conforming, it wouldn't take too long to find one that didn't validate.